scripts/s3-catalog-sync.sh

```
@@ -0,0 +1,122 @@
+fi
+
+if [[ -z "$SECRET_NAME" || "$SECRET_NAME" == "null" ]]; then
+    echo "Error: Could not extract credential secret name from DPA '$DPA_NAME'" >&2
+    exit 1
+fi
+
+echo "Extracting credentials from secret '$SECRET_NAME' (key: '$SECRET_KEY')..."
+
+CREDS=$(oc get secret "$SECRET_NAME" -n "$NAMESPACE" -o jsonpath="{.data.${SECRET_KEY}}" | base64 -d)
+
+export AWS_ACCESS_KEY_ID=$(echo "$CREDS" | grep aws_access_key_id | awk '{print $3}')
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)